src/rshim_fuse.c

```diff
@@ -1354,11 +1381,19 @@ int rshim_fuse_init(rshim_backend_t *bd)
   return 0;
 }
 
-int rshim_fuse_del(rshim_backend_t *bd)
+/*
+ * If has_misc is true, we will not delete the "misc" device file even if
+ * other device files like "boot" etc are deregistered. This allows user
+ * to send commands to the driver even if it's not fully attached to rshim
+ * device.
+ */
```

Comment:
How do we handle the rshim service stop and rshim restart?
Is it possible to remove the misc device file when rshim driver is gone?

Reply:
Yes, it should. When stopping the rshim service, `rshim_stop()` will be called, which should call `rshim_deregister(bd, false)` , where the second argument "has_misc" is different from other normal calls.

BTW, I just found I need to make additional changes for renaming "include_misc" to "has_misc". All logics have to be reversed, aka, true has to be changed to false and vice versa. I didn't do that in my previous force update.